src/stats.rs

```diff
@@ -0,0 +1,107 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::hostcalls;
+use crate::traits;
+use crate::types;
+
+pub struct Counter {
+    id: u32,
+    name: String,
+}
+
+impl Counter {
+    pub fn counter(name: String) -> Counter {
+        let returned_id = hostcalls::define_metric(types::MetricType::Counter, &name).unwrap();
+        Counter {
+            id: returned_id,
+            name,
+        }
+    }
+}
+
+impl traits::Metric for Counter {
+    fn id(&self) -> u32 {
+        self.id
+    }
+
+    fn name(&self) -> &str {
+        self.name.as_str()
+    }
+
+    fn record(&self, _: u64) {
+        // A Counter can only be incremented.
```

Comment:
Should we throw an exception here or maybe have separate trait called `MetricCounter` and place `increment` there.